### PR TITLE
Fix #1194: change max-collection/max-indexes-available setting (double up)

### LINKED ISSUE #1194: Update max-collection/max-indexes-available limits to align with Astra defaults
Currently `DatabaseLimitsConfig` set up maximum limits very low:

* Maximum 5 collections allowed
* Maximum of 50 indexes (SAI) available per Database

so let's double the defaults to align with what is typically deployed.
Note that these being config defaults, they can (and often are) overridden by ENV variables / system properties.

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Increases 2 default settings for DB config:

1. Max collections allowed from 5 to 10
2. Max indexes available per DB from 50 to 100

**Which issue(s) this PR fixes**:
Fixes #1194

**Checklist**
- [ ] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/java/io/stargate/sgv2/jsonapi/config/DatabaseLimitsConfig.java [hunk 1]
```diff
@@ -42,7 +42,7 @@ public interface DatabaseLimitsConfig {
    * Default setting for {@link #indexesAvailablePerDatabase()}: number of SAIs per Database that
    * can be created (and needed for Collection creation)
    */
-  int DEFAULT_INDEXES_AVAILABLE_PER_DATABASE = 50;
+  int DEFAULT_INDEXES_AVAILABLE_PER_DATABASE = 100;
```
[the remark this answers is not in the document]
Comment (tatu-at-datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/config/DatabaseLimitsConfig.java:45]: Ah! I did not know that. I wonder if we should instead use those values...

@jeffreyscarpenter WDYT?

[the remark this answers is not in the document]
Comment (jeffreyscarpenter) [reply src/main/java/io/stargate/sgv2/jsonapi/config/DatabaseLimitsConfig.java:45]: I agree with the change proposed in this PR. 

What Hazel is referencing is a set of changes made for deployment in a multi tenant environment. In that case we chose to increase the global limits that would be applied across all tenants to a much higher level than we would typically give an individual tenant, in order to give deference to per-tenant index limits that are enforced by the database in that deployment. 

I don't think that situation applies here. This PR is setting the Data API default values to correspond to the default index limits used by the supported backends for "self-managed" deployments (DSE 6.9 and HCD). I think this is appropriate. 

Tooling such as Mission Control that handles deployment of Data API with a supported backend should make sure to handle the interaction between the DB and Data API index limits appropriately. For example it probably makes the most sense to set the same value for each, although users could have valid reasons for setting them differently. For example a user might want to set a lower index limit on Data API vs the backend in order to make sure some indexes would be available for purposes other than Data API collections.

[the remark this answers is not in the document]
Comment (tatu-at-datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/config/DatabaseLimitsConfig.java:45]: @jeffreyscarpenter I'll go ahead with PR then.

But one note: I think overrides Hazel mentioned are per-DB, not global. They are specifically overriding values for same settings as this PR is.

**Comment by tatu-at-datastax** [comment]
@Hazel-Datastax @jeffreyscarpenter Are we good here? Regardless of deployment defaults, this helps avoid too strict settings for other use cases.

**Review by Hazel-Datastax — APPROVED** [review]

**Review by vkarpov15 — COMMENTED** [review]
Just checking that these changes will affect Astra as well? This change will be very helpful and alleviate my current biggest pain point with Data API :+1:

**Comment by tatu-at-datastax** [comment]
@vkarpov15 Sort of: there is a related change to relax actual backend guardrail limit (already merged): will only affect new DBs however. But the combination will raise limit to 10/100 as the new baseline.

File: src/main/java/io/stargate/sgv2/jsonapi/config/DatabaseLimitsConfig.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeffreyscarpenter) [reply src/main/java/io/stargate/sgv2/jsonapi/config/DatabaseLimitsConfig.java:45]: I'm not referencing Astra, what the limits are there is not relevant to this discussion. 

I'm referencing the fact that the default limit in cassandra.yaml that ships with DSE 6.9 and HCD is 100 SAI, that is what we are mapping to.

**Comment by jeffreyscarpenter** [comment]
> Just checking that these changes will affect Astra as well? This change will be very helpful and alleviate my current biggest pain point with Data API 👍

No, they don't affect Astra. That is configured separately.

File: src/main/java/io/stargate/sgv2/jsonapi/config/DatabaseLimitsConfig.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (tatu-at-datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/config/DatabaseLimitsConfig.java:45]: Ok. That makes sense. I was thinking of earlier Astra limits that just happened to coincide so I jumped to (wrong) conclusions.

**Pull request merged**